src/cargo/ops/registry.rs

```diff
@@ -984,6 +984,14 @@ pub fn search(
         )
     })?;
 
+    if total_crates == 0 {
+        return Err(anyhow!(
+            "could not find any crates matching `{}` from the registry at {}",
+            query,
+            registry.host()
+        ));
+    }
```

Comment:
Could you expand on why you feel this behavior change is acceptable?

Reply:
I'm not entirely certain that an error is the correct answer, I mostly did that as it was requested. I think an error might be too harsh since there really wasn't an "error", just no matches. Keeping that in mind, an error is the easiest way to change the output code (1.61.0 does allow `main` to return different codes which could be a solution).

That being said, something should probably be shown when no crates were found. Having it output nothing seems like a misstep.

Reply:
> 1.61.0 does allow main to return different codes which could be a solution).

There is also `std::process::exit` (generally not good to do in a library call though)

> Having it output nothing seems like a misstep.

At least `apt search` doesn't really say anything.